src/peergos/shared/crypto/FragmentedPaddedCipherText.java

```diff
@@ -132,9 +132,13 @@ public static FragmentedPaddedCipherText fromCbor(Cborable cbor) {
                                                   NetworkAccess network,
                                                   ProgressConsumer<Long> monitor) {
         if (inlinedCipherText.isPresent()) {
-            if (header.isPresent())
-                return Futures.of(new CipherText(nonce, ArrayOps.concat(header.get(), inlinedCipherText.get())).decrypt(from, fromCbor));
-            return Futures.of(new CipherText(nonce, inlinedCipherText.get()).decrypt(from, fromCbor));
+            T cipherText = header.isPresent() ?
+                    new CipherText(nonce, ArrayOps.concat(header.get(), inlinedCipherText.get())).decrypt(from, fromCbor)
+                    : new CipherText(nonce, inlinedCipherText.get()).decrypt(from, fromCbor);
+            if (cipherText instanceof byte[]) {
+                monitor.accept((long) ((byte[]) (cipherText)).length);
```

Comment:
Can we do this without casts and instanceof?

Reply:
It's probably okay to just use the length of the ciphertext here. As we know it will complete the download regardless.

Reply:
> Can we do this without casts and instanceof?

No. Could be of type CryptreeNode.ChildrenLinks

Reply:
> It's probably okay to just use the length of the ciphertext here. As we know it will complete the download regardless.

I looked at the length of various structures. Uness i'm looking at the wrong one, the length does not match ((byte[]) (cipherText)).length

Reply:
The ciphertext is strictly longer yes. A better solution would be to add a decrypt method that takes the progress counter.